crates/ty_project/src/walk.rs

```diff
@@ -232,8 +237,12 @@ impl<'a> ProjectFilesWalker<'a> {
                                 }
                             }
 
-                            let mut paths = paths.lock().unwrap();
-                            paths.push(entry.into_path());
+                            // If this returns `Err`, then the file was deleted between now and when the walk callback was called.
+                            // We can ignore this.
+                            if let Ok(metadata) = system.path_metadata(entry.path()) {
```

Comment:
I was wondering why you can't use `ignore::DirEntry::file_type` here, and I think it's because you also need the file permissions? Because directory entries (on Unix) will include the `FileType` free of charge. But not permissions.

Reply:
We need permissions and, more importantly, the last modified time